doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+
+    sample property
+    sample properties
+        TODO
+
+    scikit-learn-contrib
+        TODO
+
+    scorer
+        TODO
+        See :ref:`scoring_parameter`.
+        See also :term:`evaluation metric`.
+
+    sparse matrix
+        TODO
+
+    target
+    targets
+        TODO
+
+        Dependent variable or outcome variable.
+
+    unlabeled
+    unlabeled data
+        TODO
+
+Class APIs and Estimator Types
+==============================
+
+.. glossary::
+
+    classifier
+        TODO
+        Mention that within scikit-learn, all support multi-class
+        classification, defaulting to OvR.
+        Mention :func:`~base.is_classifier`.
+
+    clusterer
+        TODO
+
+    cross validation splitter
+    CV splitter
+        TODO
+
+    estimator
+        TODO
+
+        The core functionality of some estimators may also be available as a
+        :term:`function`.
+
+    feature extractor
+        A :term:`tranformer` which takes input where each sample is not
+        represented as an :term:`array-like` object of fixed length, and
+        produces an `array-like` object of :term:`features` for each sample
+        (and thus a 2-dimensional array-like for a set of samples).  In other
+        words, it (lossily) maps a non-rectangular data representation into
+        :term:`rectangular` data.
+
+    meta-estimator
+        TODO
+
+        Mention duck typing. Mention that duck typing of methods only works
+        after fitting. Mention lenient validation. ?Mention sample props.
+        Mention clone.
+
+    outlier detector
+        TODO
+
+    predictor
+        An :term:`estimator` which provides :term:`predict`.
+        This encompasses :term:`classifier`, :term:`regressor`,
+        :term:`outlier detector` and sometimes :term:`clusterer` (at least when
+        they are inductive).  In scikit-learn, if an estimator is not a
+        predictor, it is usually a :term:`transformer`.
+
+    regressor
+        TODO
+        Mention :func:`~base.is_regressor`.
+
+    transformer
+        TODO
+
+    vectorizer
+        See :term:`feature extractor`.
+
+    ``Xt``
+        Shorthand for transformed ``X``.
+
+There are further APIs specifically related to a small family of estimators,
+such as:
+
+* :class:`neighbors.DistanceMetric`
+* :class:`gaussian_process.kernels.Kernel`
+* ``tree.Criterion``
+
+.. _glossary_target_types:
+
+Target Types
+============
+
+.. glossary::
+
+    binary
+        A classification problem consisting of two classes.  A binary target
+        may represented as for a :term:`multiclass` problem but with only two
+        labels.  A binary decision function is represented as a 1d array.
+
+        Note that a dataset sampled from a multiclass ``y`` or a continuous
+        ``y`` may appear to be binary.
+
+        :func:`~utils.multiclass.type_of_target` will return 'binary' for
+        binary input, or a similar array with only a single class present.
```

Comment:
decision function is special for this one! And a bunch of metrics (binary f1 etc) are, too!

Reply:
Oh and I think the lexicographic smallest class is the positive class for binary, right? Actually there is no test for that, we only have a test for labels 0 and 1, and there we test that 1 is the positive class... opened #10175

Reply:
I think you mean the lexicographic greatest class is the positive class....

Reply:
yes, indeed. I was confused for a bit. Is that documented somewhere?